tools/runqslower.py

```diff
@@ -182,12 +182,19 @@
     struct task_struct *prev = (struct task_struct *)ctx->args[1];
     struct task_struct *next= (struct task_struct *)ctx->args[2];
     u32 tgid, pid, prev_pid;
-    long state;
+    long task_state;
 
     // ivcsw: treat like an enqueue event and store timestamp
-    bpf_probe_read_kernel(&state, sizeof(long), (const void *)&prev->state);
+//kernel>=5.14, tsk->state turn to tsk->__state                                                                              
+#if LINUX_VERSION_MAJOR>=5 && LINUX_VERSION_PATCHLEVEL>=14
+#define state __state
+#else
+#define state state
+#endif
+    bpf_probe_read_kernel(&task_state, sizeof(long), (const void *)&prev->state);
```

Comment:
I'm suffering this issue too :( cannot execute runqslower/runqlat on v5.15-rc7.

This is part of diff that changed task_struct's state field:
297 -       volatile long                   state;
298 +       unsigned int                    __state;

it would be better your commit reflect its type change too.

Reply:
If you have latest kernel, please use [libbpf-tools](https://github.com/iovisor/bcc/blob/master/libbpf-tools/runqlat.bpf.c). They have the same functionality as the BCC one.